### ISSUE TITLE
Fetching of grades from some courses failed

### ISSUE THREAD

**Issue by ppv1979** [mudrd8mz/moodle-mod_subcourse#38]
In several courses we see the following situation: Fetching fails with the result like this:

Debug info:
Column 'hidden' cannot be null
UPDATE mdl_main_grade_grades SET itemid = ?,userid = ?,rawgrade = ?,rawgrademax = ?,rawgrademin = ?,rawscaleid = ?,usermodified = ?,finalgrade = ?,hidden = ?,locked = ?,locktime = ?,exported = ?,overridden = ?,excluded = ?,timecreated = ?,timemodified = ?,aggregationstatus = ?,aggregationweight = ?,feedback = ?,feedbackformat = ?,information = ?,informationformat = ? WHERE id=?
[array (
  0 => '2990',
  1 => '116',
  2 => 80.0,
  3 => 100.0,
  4 => 0.0,
  5 => NULL,
  6 => '85',
  7 => 80.0,
  8 => NULL,
  9 => '0',
  10 => '0',
  11 => '0',
  12 => '0',
  13 => '0',
  14 => NULL,
  15 => '1452446405',
  16 => 'used',
  17 => '0.06667',
  18 => NULL,
  19 => '0',
  20 => NULL,
  21 => '0',
  22 => '240505',
)]
Backtrace:
* line 1561 of /lib/dml/mysqli_native_moodle_database.php: call to moodle_database->query_end()
* line 1593 of /lib/dml/mysqli_native_moodle_database.php: call to mysqli_native_moodle_database->update_record_raw()
* line 253 of /lib/grade/grade_object.php: call to mysqli_native_moodle_database->update_record()
* line 1040 of /lib/grade/grade_grade.php: call to grade_object->update()
* line 648 of /lib/grade/grade_grade.php: call to grade_grade->update()
* line 241 of /mod/subcourse/locallib.php: call to grade_grade->set_hidden()
* line 75 of /mod/subcourse/classes/task/fetch_grades.php: call to subcourse_grades_update()
* line 201 of /lib/cronlib.php: call to mod_subcourse\task\fetch_grades->execute()
* line 119 of /lib/cronlib.php: call to cron_run_inner_scheduled_task()
* line 73 of /lib/cronlib.php: call to cron_run_scheduled_tasks()
* line 61 of /admin/cli/cron.php: call to cron_run()

The userid in such queries is the userid of user with role 'teacher' in the particular course (we can have more than one teacher in a course but I am not sure that it has any influence). Do you have any ideas about how to fix this? 

Moodle 3.8.4+ (Build: 20200730)
mod_subcourse 8.0.0 (2020071100)

**Comment by mudrd8mz** [comment]
Hmm. I admit the code does not expect that it would see the remote grade with the `hidden` property set to NULL. Everything I've seen so far suggested that this would not be the case.

Are you able to run the following query for me in your database so that we can potentially debug where these NULLs are coming from?

```
SELECT id, itemid, timecreated FROM mdl_grade_grades WHERE hidden IS NULL;
```

**Comment by ppv1979** [comment]
@mudrd8mz, I've run the query — 0 rows.

**Comment by mudrd8mz** [comment]
I'm on the track now. I can reproduce the problem and I believe I already know what is going on.

**Comment by mudrd8mz** [comment]
3afd547b5c140fa024cfb8d365dc52f80c17daeb should be the right for this. Just finishing some more tests and will release a new version 8.0.1 to address this.

**Comment by mudrd8mz** [comment]
Fixed in v8.0.1

Thanks again for raising this.

**Comment by ppv1979** [comment]
@mudrd8mz thank you for the fix!
